products/memgraph/upcoming/tutorials/replication.md

```diff
@@ -0,0 +1,135 @@
+
+### Introduction
+
+This example demonstrates how to create a simple cluster of nodes running
+Memgraph instances, and set up replication. We'll demonstrate the use and
+behavior of the available synchronization modes.
+
+### Cluster topology
+
+The cluster consists of four nodes, a single main and three replicas:
+
+* the main node, containing the original data to be replicated to other nodes,
+* a node that will be replicated to using the sync mode,
+* a node that will be replicated to using the async mode,
+* a node that will be replicated to using the sync-with-timeout mode.
+
+We'll be hitting the main node with create queries, creating nodes and edges,
+while running BFS on the replicas, and observing the effects of different sync
+modes used.
+
+### Configuring the cluster
+
+We'll use Docker to set up and run the cluster on your local machine, so make
+sure you have it installed and ready, and grab your Memgraph docker image.
+Look [here](https://docs.memgraph.com/memgraph/getting-started/installation/docker-installation) for instructions.
+In addition, make sure you have either [mgclient](https://github.com/memgraph/mgclient) or [mgconsole](https://github.com/memgraph/mgconsole) installed.
+
+We have to setup the cluster nodes' ports properly, so we'll use the following
+mapping:
+* the main node will have port 7687 assigned
+* the sync node will have port 7688 assigned
+* the async node will have port 7689 assigned
+* the sync-with-timeout node will have port 7690 assigned.
+Note that we could've chosen arbitrary ports.
+
+We fire up the terminal, and for each Memgraph instance (node)  we have to start, we'll
+run:
+
+```plaintext
+docker run --rm -p PORT:7687 memgraph:1.3.0-enterprise
+```
+
+where PORT runs from 7687 to 7690. The above commands will start a Memgraph
+node, and map it to the appropriate port on your host machine.
+
+Now, to set up the cluster, we'll have to start an mg_client or mgconsole
+instance for every running Memgraph node, and connect to it. To do this, we
+have to figure out their IP addresses. Running
+
+```plaintext
+docker ps
+```
+
+will list all the running docker processes, along with their "CONTAINER ID" and
+"NAME". Running
+
+```plaintext
+docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' container_name_or_id
+```
+
+where "container_name_or_id" is either the "CONTAINER ID" or "NAME" listed by
+`docker ps`, we can get the IP addresses of all the nodes. Once we have them, we
+can start setting up the replication. Let's say the ip addresses are as follows:
+* main:                      172.17.0.2
+* sync replica:              172.17.0.3
+* async replica:             172.17.0.4
+* sync-with-timeout replica: 172.17.0.5 .
+
+Let's assume we're using mgconsole to connect to and query the nodes. Firstly,
+we have to set up the replicas. We connect to a replica by running
+
+```plaintext
+mgconsole --host REPLICA_IP_ADDRESS --port REPLICA_PORT --use-ssl=false
```

Comment:
So it doesn't work if you remove all -p PORT:7687 from each docker run and use 7687 as the port but with different host addresses?

Reply:
Actually, it does! Thanks, nice catch!